### DISCUSSION TITLE
Any way to have cameras selectively render things?

### DISCUSSION THREAD

**Discussion by piboistudios** [HaxeFlixel/flixel#2607] [Q&A] [answered]
I have a turn based game running on a server that then records GIFs of the battle and sends replays to chatrooms (like Slack, Discord, etc..)

I can add multiple cameras with FlxG to render/record multiple battle scenes concurrently, but is it possible to utilize the exact same battle scene, but show different characters/monsters/fx for different cameras? e.g. Only show certain FlxGroups to certain cameras

**Comment by piboistudios** [comment]
Nevermind, see: https://stackoverflow.com/a/8176178/1538301

**Reply by Geokureli** [reply]
worth mentioning that this the article you linked in almost 11 years old and refers to the Action Script version of flixel. the following:
> The biggest frustration: Currently FlxGroup does not pass its cameras on to its members. Hopefully this will be added into future versions of Flixel

This is not true, anymore. you can set a group's `cameras` and it will affect all of it's member's `cameras` (unless that member's `cameras` or `camera` field was specifically set). In fact I recommend doing this in most cases

**Comment by Geokureli** [comment, marked as answer]
All sprites will, by default, draw to all cameras set as a [default draw target](https://api.haxeflixel.com/flixel/system/frontEnds/CameraFrontEnd.html#setDefaultDrawTarget), to only have certain cameras draw a sprite you must set either it's `cameras` or `camera` (which just sets `cameras = [cam]`) fields.

`FlxGroups` have a `cameras` field as well, setting it will set the default `cameras` of any child sprite it contains. However, any child sprite of that group that has it's own cameras set will draw to those cameras instead of the group's.

Note: I wanted wrote a separate answer so that future people searching this aren't deferred to a ~11 year old stack overflow link.
